docs/ci.md

```diff
@@ -54,3 +54,201 @@ Received 0 of 614674276 (0.0%), 0.0 MBs/sec
 ...
 ```
 
+### Failing tests
+
+At times, there will be tests which only fail intermittently. It's critical to determine if these
+are real bugs or if they are simply flaky tests. Your first approach should be to reproduce the
+failure locally in the simplest way.
+
+#### Reproduce locally
+
+Here are some tips to do so:
+
+- Open IntelliJ, and run the test.
+- If it fails, great, we know it's reproducible locally.
+- If it does not fail,
+  [edit the run configuration](https://www.jetbrains.com/help/idea/run-debug-configuration.html)
+  that was just created (it will be named after the test), and make
+  it [repeat until the test fails](https://www.jetbrains.com/help/idea/run-debug-configuration-junit.html#tests)
+  .
+
+If you cannot reproduce it locally, the next step is to reproduce it in our CI environment.
+
+#### Reproduce in CI
+
+The first thing we'll do is create a branch so we can run everything in isolation.
+
+##### Isolate test environment
+
+```shell
+git checkout -b 12983-flaky-test-issue-repro
+git push --set-upstream origin 12983-flaky-test-issue-repro
+```
+
+This will also allow you to
+run [the CI workflow](https://github.com/camunda/zeebe/actions/workflows/ci.yml) for that specific
+branch as many times as you want. If you don't know how to do this, you can read up on
+it [here](https://docs.github.com/en/actions/managing-workflow-runs/manually-running-a-workflow).
+
+##### Run a single job
+
+The next step is to narrow down the workflow to our failing test. This will make the feedback loop
+shorter, and minimize resource usage during investigation.
+
+> NOTE: it can happen that a flaky test is only flaky when there is pressure on the system, so we
```

Comment:
```suggestion
> **Note**
> It can happen that a flaky test is only flaky when there is pressure on the system, so we
```
See: https://github.com/community/community/discussions/16925